app/validation/mandatory_check.py

```diff
@@ -16,16 +16,21 @@ def validate(self, user_answer):
 
         if isinstance(user_answer, list):
             for answer in user_answer:
-                if answer:
+                if answer and not str(answer).isspace():
                     validation_result.is_valid = True
                     return validation_result
-            else:
-                validation_result.is_valid = False
-                validation_result.errors.append(AbstractValidator.MANDATORY)
+                else:
+                    validation_result.is_valid = False
+                    # We do not return at this point as there may still be
+                    # a valid response in the list
+
+            # No valid values in list
+            validation_result.errors.append(AbstractValidator.MANDATORY)
         else:
             if user_answer and not str(user_answer).isspace():
                 validation_result.is_valid = True
             else:
                 validation_result.is_valid = False
                 validation_result.errors.append(AbstractValidator.MANDATORY)
+
```

Comment:
There is duplicated functionality in this method, I think it would be better to extract a method and have the loop repeatedly call it. For example:

```
    def validate(self, user_answer):
        if isinstance(user_answer, list):
            for answer in user_answer:
                validation_result = self._validate(answer)
                if validation_result.is_valid:
                    break
        else:
            validation_result = self._validate(user_answer)

        return validation_result

    def _validate(self, user_answer):
        validation_result = ValidationResult()

        if user_answer and not str(user_answer).isspace():
            validation_result.is_valid = True
        else:
            validation_result.is_valid = False
            validation_result.errors.append(AbstractValidator.MANDATORY)

        return validation_result
```